net/avahi/Portfile

```diff
@@ -89,33 +89,26 @@ configure.args      --disable-autoipd \
 # __APPLE_USE_RFC_2292 should be removed once avhi is updated to support RFC 3542
 configure.cppflags-append   -L${prefix}/lib -D__APPLE_USE_RFC_2292
 
-set launch_daemons {org.freedesktop.avahi-daemon.plist org.freedesktop.avahi-dnsconfd.plist}
-
-post-destroot {
-    xinstall -d -m 0755 ${destroot}${prefix}/Library/LaunchDaemons
-    foreach fl ${launch_daemons} {
-        file attributes ${destroot}/Library/LaunchDaemons/${fl} -permissions 0644
-        move ${destroot}/Library/LaunchDaemons/${fl} ${destroot}${prefix}/Library/LaunchDaemons
-    }
-
-    # previous command should move all LaunchDaemons
-    foreach fl [glob -nocomplain -tails -directory ${destroot}/Library/LaunchDaemons *] {
-        ui_error "${destroot}/Library/LaunchDaemons should be empty"
-        ui_error "${fl} should not exist in ${destroot}/Library/LaunchDaemons"
-        return -code error "unaccounted for file in LaunchDaemons"
-    }
+set daemon_uniquename   org.freedesktop.avahi-daemon
+set dnsconfd_uniquename org.freedesktop.avahi-dnsconfd
 
-    if {${startupitem.install}} {
-        foreach fl ${launch_daemons} {
-            ln -s ${prefix}/Library/LaunchDaemons/${fl} ${destroot}/Library/LaunchDaemons/${fl}
-        }
-    }
-}
+startupitem.type    launchd
+startupitem.create  no
+startupitems        name        avahi-daemon \
+                    location    LaunchDaemons \
+                    uniquename  ${daemon_uniquename} \
+                    plist       ${daemon_uniquename}.plist \
+                    name        avahi-dnsconfd \
+                    location    LaunchDaemons \
+                    uniquename  ${dnsconfd_uniquename} \
+                    plist       ${dnsconfd_uniquename}.plist
 
-post-activate {
-    if {${startupitem.install}} {
-        foreach fl ${launch_daemons} {
-            file attributes ${prefix}/Library/LaunchDaemons/${fl} -owner root -group wheel
+post-destroot {
+    foreach uniquename "${daemon_uniquename} ${dnsconfd_uniquename}" {
+        xinstall -d -m 0755 ${destroot}${prefix}/etc/LaunchDaemons/${uniquename}
+        move ${destroot}/Library/LaunchDaemons/${uniquename}.plist ${destroot}${prefix}/etc/LaunchDaemons/${uniquename}/
+        if {${startupitem.install} && [geteuid] == 0} {
+            ln -s ${prefix}/etc/LaunchDaemons/${uniquename}/${uniquename}.plist ${destroot}/Library/LaunchDaemons/
         }
```

Comment:
You need an else clause here that creates the symlink in `${destroot}${prefix}/etc/LaunchDaemons`. Otherwise `port load` won't work for users that have set `startupitem_install no`.